lib/dependabot/file_updaters/java_script/npm_and_yarn/npm_lockfile_updater.rb

```diff
@@ -311,6 +336,29 @@ def git_ssh_requirements_to_swap
             @git_ssh_requirements_to_swap
           end
 
+          def prepared_npm_lockfile_content(content)
+            JSON.dump(
+              remove_dependency_from_npm_lockfile(parsed_package_lock(content))
+            )
+          end
+
+          def remove_dependency_from_npm_lockfile(npm_lockfile)
+            dependencies = npm_lockfile["dependencies"]&.
+                           reject { |key, _| key == dependency.name }&.
+                           map do |key, value|
+                             [key, remove_dependency_from_npm_lockfile(value)]
+                           end&.to_h
+            npm_lockfile.merge("dependencies" => dependencies)
+          end
+
+          def parsed_package_lock(content)
+            return {} unless content
+
+            JSON.parse(content)
+          rescue JSON::ParserError
```

Comment:
No need to rescue here (or to handle the no content case - we should blow up in both cases.